components/ai_chat/core/common/prefs.cc

```diff
@@ -7,13 +7,105 @@
 
 #include <utility>
 
+#include "base/json/values_util.h"
+#include "base/strings/string_util.h"
+#include "base/time/time.h"
+#include "base/uuid.h"
+#include "base/values.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
 #include "brave/components/ai_chat/core/common/mojom/customization_settings.mojom.h"
 #include "brave/components/ai_chat/core/common/pref_names.h"
 #include "components/prefs/pref_service.h"
 #include "components/prefs/scoped_user_pref_update.h"
 
 namespace ai_chat::prefs {
 
+namespace {
+
+// Helper function to convert a dictionary to SmartMode
+mojom::SmartModePtr SmartModeDictToStruct(const std::string& id,
+                                          const base::Value::Dict& mode_dict) {
+  const std::string* shortcut = mode_dict.FindString("shortcut");
+  const std::string* prompt = mode_dict.FindString("prompt");
+  const std::string* model = mode_dict.FindString("model");
+
+  if (!shortcut || !prompt) {
+    return nullptr;
+  }
+
+  // Parse timestamps using base::ValueToTime
+  base::Time created_time, last_used;
+  if (const base::Value* created_time_value = mode_dict.Find("created_time")) {
+    auto parsed_time = base::ValueToTime(*created_time_value);
+    if (parsed_time) {
+      created_time = *parsed_time;
+    }
+  }
+  if (const base::Value* last_used_value = mode_dict.Find("last_used")) {
+    auto parsed_time = base::ValueToTime(*last_used_value);
+    if (parsed_time) {
+      last_used = *parsed_time;
+    }
+  }
+
+  return mojom::SmartMode::New(
+      id, *shortcut, *prompt, model ? std::make_optional(*model) : std::nullopt,
+      created_time, last_used);
+}
+
+// Helper function to convert SmartMode to dictionary
+base::Value::Dict SmartModeStructToDict(const mojom::SmartModePtr& smart_mode) {
+  base::Value::Dict mode_dict;
+  mode_dict.Set("shortcut", smart_mode->shortcut);
+  mode_dict.Set("prompt", smart_mode->prompt);
+  if (smart_mode->model) {
+    mode_dict.Set("model", *smart_mode->model);
+  }
+  mode_dict.Set("created_time", base::TimeToValue(smart_mode->created_time));
+  mode_dict.Set("last_used", base::TimeToValue(smart_mode->last_used));
+  return mode_dict;
+}
+
+bool IsValidAndUniqueShortcut(const PrefService& prefs,
+                              const std::string& shortcut,
+                              const std::string& exclude_id) {
+  // Check format validation
+  if (shortcut.empty()) {
+    return false;
+  }
+
+  if (!std::all_of(shortcut.begin(), shortcut.end(), [](char c) -> bool {
+        return base::IsAsciiAlphaNumeric(c) || c == '_' || c == '-';
+      })) {
+    return false;
```

Comment:
Seems like we don't have unit test case cover this

Reply:
added UpdateSmartModeToPrefs_InvalidShortcut in https://github.com/brave/brave-core/pull/31422/commits/c08ed6c623230b6affe73e51af28e6c265ec9690